airflow/auth/managers/base_auth_manager.py

```diff
@@ -33,3 +33,8 @@ class BaseAuthManager(LoggingMixin):
     def get_user_name(self) -> str:
         """Return the username associated to the user in session."""
         ...
+
+    @abstractmethod
+    def is_logged_in(self) -> bool:
+        """Return whether the user is logged in."""
+        ...
```

Comment:
should this be abstract ? Or maybe we could implement it using `get_user_name` ? If it returns None if user is not logged ?

Reply:
I dont really like that. Each auth manager can be very different and retrieving the user name to check whether the user is logged in feels wrong to me. Some auth manager can have very complex/costly operation to retrieve the username and some cheap one to just check whether a user is logged in, so I rather have two operation. If one auth manager thinks one method can use the other, it can do it in their own implementation

Reply:
ok gotcha